lib/kernel.js

```diff
@@ -86,36 +141,53 @@ class MiddlewareAdapter implements HydrogenKernelMiddlewareThunk {
     callWatches: boolean,
     onResults: ResultsCallback
   ): void {
+    // We don't want to repeatedly wrap the onResults callback every time we
+    // fall through, but we need to do it at least once before delegating to
+    // the KernelTransport.
+    let safeOnResults =
+      this._middleware.execute || this._next instanceof KernelTransport
+        ? protectFromInvalidMessages(onResults)
+        : onResults;
+
```

Comment:
Note that the more straightforward code:
```js
if (this._middleware.execute) {
  this._middleware.execute(..., protectFromInvalidMessages(onResults));
} else if (this._next instanceof KernelTransport) {
  this._next.execute(..., protectFromInvalidMessages(onResults));
} else {
  this._next.execute(..., onResults);
}
```

does not pass flow. Flow thinks that `protectFromInvalidMessages` is capable of deleting `this._middleware.execute`.

I've settled on this approach over a `$FlowFixMe` because the `$FlowFixMe` would be overly-broad and may prevent flow from catching real errors.